kikuchipy/signals/ebsd_master_pattern.py

```diff
@@ -75,3 +227,241 @@ class LazyEBSDMasterPattern(EBSDMasterPattern, LazySignal2D):
+    niip = nii + 1
+    nijp = nij + 1
+    niip = np.where(niip < npx, niip, nii).astype(int)
+    nijp = np.where(nijp < npy, nijp, nij).astype(int)
+    nii = np.where(nii < 0, niip, nii).astype(int)
+    nij = np.where(nij < 0, nijp, nij).astype(int)
+    di = i - nii + scale
+    dj = j - nij + scale
+    dim = 1.0 - di
+    djm = 1.0 - dj
+
+    return nii, nij, niip, nijp, di, dj, dim, djm
+
+
+def _get_patterns_chunk(
+    r: Rotation,
+    dc: Vector3d,
+    master_north: np.ndarray,
+    master_south: np.ndarray,
+    npx: int,
+    npy: int,
+    rescale: bool,
+    dtype_out: Optional[np.dtype] = np.float32,
+) -> np.ndarray:
+    """Get the EBSD patterns on the detector for each rotation in the chunk.
+    Each pattern is found by a bi-quadratic interpolation of the master pattern
+    as described in EMsoft.
+
+    Parameters
+    ----------
+    r : Rotation
+        Rotation object with all the rotations for a given chunk.
+    dc : Vector3d
+        Direction cosines unit vector between detector and sample.
+    master_north : numpy.ndarray
+        Northern hemisphere of the master pattern.
+    master_south : numpy.ndarray
+        Southern hemisphere of the master pattern.
+    npx : int
+        Number of pixels in the x-direction on the master pattern.
+    npy: int
+        Number of pixels in the y-direction on the master pattern.
+    rescale : bool
+        Whether to call rescale_intensities() or not.
+    dtype_out : numpy.dtype, optional
+        Data type of the returned patterns, by default np.float32.
+
+    Returns
+    ----------
+    numpy.ndarray
+        (n, y, x) array containing all the simulated patterns.
+    """
+    m = r.shape[0]
+    simulated = np.empty(shape=(m,) + dc.shape, dtype=dtype_out)
+    scale_factor = (npx - 1) / 2
+    for i in range(m):
+        rot_dc = Rotation(r[i]) * dc
+        (
+            nii,
+            nij,
+            niip,
+            nijp,
+            di,
+            dj,
+            dim,
+            djm,
+        ) = _get_lambert_interpolation_parameters(
+            rotated_direction_cosines=rot_dc,
+            scale=scale_factor,
+            npx=npx,
+            npy=npy,
+        )
+        pattern = np.where(
+            rot_dc.z >= 0,
+            (
+                master_north[nii, nij] * dim * djm
+                + master_north[niip, nij] * di * djm
+                + master_north[nii, nijp] * dim * dj
+                + master_north[niip, nijp] * di * dj
+            ),
+            (
+                master_south[nii, nij] * dim * djm
+                + master_south[niip, nij] * di * djm
+                + master_south[nii, nijp] * dim * dj
+                + master_south[niip, nijp] * di * dj
+            ),
+        )
+        if rescale:
+            pattern = rescale_intensity(pattern, dtype_out=dtype_out)
+        simulated[i] = pattern
+    return simulated
+
+
+def _min_number_of_chunks(
+    d: EBSDDetector, r: Rotation, dtype_out: np.dtype
```

Comment:
I suggest that this function takes in a tuple and an integer instead of the detector and rotation objects.